ekncontent/ekncontent/eknc-domain.c

```diff
@@ -413,8 +415,15 @@ eknc_domain_initable_init (GInitable *initable,
       /* Find out root relative path from manifest file */
       g_autoptr(GFile) p = g_file_get_parent (self->manifest_file);
       g_autoptr(GString) relative_path = g_string_new ("");
-      while ((p = g_file_get_parent (p)))
-        g_string_append (relative_path, "../");
+      while (TRUE)
+        {
+          GFile *old_p = g_steal_pointer (&p);
```

Comment:
This could be:

```cpp
while (TRUE)
  {
    g_autoptr(GFile) old_p = g_steal_pointer (&p);
    p = g_file_get_parent (old_p);
    if (p == NULL)
      break;
  }
```